### DISCUSSION TITLE
Region lifetime

### DISCUSSION THREAD

**Discussion by Tum4ik** [PrismLibrary/Prism#2458] [Xamarin.Forms] [answered]
I heard at some Brian's videos that region is created once requested and then Prism just changes regions when user navigates, but does not recreate them.
But in Xamarin.Forms I experience the behavior that the region is always recreated when requested. So, I guess may be those Brian's statement is related only to WPF.
Is there some way to change that behavior or it is the only possible approach?

**Comment by dansiegel** [comment]
I believe you're getting a little confused here... If a view (for simplicity we'll say a Page, but could be literally any View) contains a child which added the `RegionManager.RegionName` BindableProperty then it will "Create" the Region immediately. Otherwise when you go to Navigate the Region wouldn't exist in the RegionManager and your RequestNavigate would fail.

That said no child content is necessarily added to the Region until you call RequestNavigate. In Prism 8.1 which was released yesterday there are some new APIs that do make it possible to register Views for Regions so that when a Region is created then one or more Views can be added to the Region immediately. Prism.Forms.Regions does contain a bunch of tests which you can use for guidance as well as by looking at the e2e app's HelloRegions module.

**Reply by Tum4ik** [reply]
Oh... Now I understand I've even confused you @dansiegel )) This question is mostly about View lifetime which I put into the region. So, I think the question should look like:

I heard at some Brian's videos that **view for region navigation** is created once requested and then Prism just changes **views** when user navigates, but does not recreate them.
But in Xamarin.Forms I experience the behavior that the **view** is always recreated when requested. So, I guess may be those Brian's statement is related only to WPF.
Is there some way to change that behavior or it is the only possible approach?

**Comment by Tum4ik** [comment, marked as answer]
I have found the answer on my question. I should use `IRegionAware` interface for my region view model and return `true` from the `IsNavigationTarget` method. 
```csharp
public bool IsNavigationTarget(INavigationContext navigationContext)
{
    return true;
}
```
Then Prism stops to create new instance of the region view model each time I navigate to it.
